ServiceDesk/ServiceDeskClient.cs

```diff
@@ -33,42 +34,34 @@ public class ServiceDeskClient : IServiceDeskClient
                 },
             };
 
-        public ServiceDeskClient(IHttpClientFactory clientFactory)
+        public ServiceDeskClient(IHttpClientFactory clientFactory, ServiceDeskConfiguration serviceDeskConfiguration)
         {
-            this.ClientFactory = clientFactory;
+            this.clientFactory = clientFactory;
+            this.serviceDeskConfiguration = serviceDeskConfiguration;
         }
 
-        private IHttpClientFactory ClientFactory { get; }
-
-        public async Task<IEnumerable<ServiceDeskRequestTypeDTO>> GetRequestTypes(CancellationToken cancellationToken)
+        public Task<IEnumerable<ServiceDeskRequestTypeDTO>> GetRequestTypes(CancellationToken cancellationToken)
         {
-            var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestTypesUrl);
-
-            httpRequest.Headers.Add("x-api-key", "not-installed");
-
-            var client = this.ClientFactory.CreateClient();
-
-            var response = await client.SendAsync(httpRequest, cancellationToken);
-
-            var responseBody = await response.Content.ReadAsStreamAsync();
-
-            var options = new JsonSerializerOptions
-            {
-                PropertyNameCaseInsensitive = true,
-            };
+            return this.GetByUrl<ServiceDeskRequestTypeDTO>($"{this.serviceDeskConfiguration.ApiUrl}{requestTypesUrl}", cancellationToken);
+        }
 
-            var serviceDeskRequestTypes = await JsonSerializer.DeserializeAsync<IEnumerable<ServiceDeskRequestTypeDTO>>(responseBody, options);
+        public Task<IEnumerable<ServiceDeskRequestDTO>> GetCurrentRequests(string username, CancellationToken cancellationToken)
+        {
+            return this.GetByUrl<ServiceDeskRequestDTO>($"{this.serviceDeskConfiguration.ApiUrl}{currentRequestsUrl}{username}", cancellationToken);
+        }
 
-            return serviceDeskRequestTypes;
+        public Task<IEnumerable<ServiceDeskRequestPriorityDTO>> GetPriorities()
+        {
+            return Task.FromResult(this.priorities);
         }
 
-        public async Task<IEnumerable<ServiceDeskRequestDTO>> GetCurrentRequests(string username, CancellationToken cancellationToken)
+        public async Task<IEnumerable<T>> GetByUrl<T>(string url, CancellationToken cancellationToken)
         {
-            var httpRequest = new HttpRequestMessage(HttpMethod.Get, $"{currentRequestsUri}{username}");
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, url);
 
             httpRequest.Headers.Add("x-api-key", "not-installed");
```

Comment:
This should also be configured in application settings.

Reply:
Fixed